### PR TITLE
Add support for MinMax scaling

### PR DESCRIPTION
**Pull request by yoziru**
Resolves https://github.com/facebook/prophet/issues/2437
This PR adds a new `scaling` parameter, with `minmax` scaling support (on top of the default `absmax` scaling in place right now). This is useful for certain datasets where `AbsMax` will compress the range (see https://github.com/facebook/prophet/issues/2437) to be `[0.99999... - 1.0]`, leading to bad fitting of parameters. This dataset is now added as a test in `data3.csv`.

For now the default `scaling` remains `absmax`, and de-serialization is backwards-compatible.

Here is how this looks like for the 3 test datasets.

| AbsMax | MinMax|
|---|---|
| output-data-absmax | output-data-minmax |
| output-data2-absmax | output-data2-minmax |
| output-data3-absmax | output-data3-minmax |

### CODE REVIEW

**Comment by tcuongd** [comment]
This seems like a great contribution, thanks so much! I'll take a proper look soon.

**Comment by yoziru** [comment]
@tcuongd re this change, settled on this approach in the PR because it has the most minimal changes to the code (and fully backwards-compat).

An alternative was to have a more generic scaler implementation so it can support other types of scalers or even user-defined funcs, but not sure if we need to go there. Wdyt?

File: python/prophet/serialize.py [hunk 1]
```diff
@@ -144,6 +144,23 @@ def model_from_dict(model_dict):
...
+            if attribute == 'train_component_cols':
+                # Special handling because of named index column
+                df.columns.name = 'component'
+                df.index.name = 'col'
+            setattr(model, attribute, df)
+    
+    # backwards compat with prophet <=1.1.4
+    if 'scaling' not in model_dict:
+        model_dict['scaling'] = 'absmax'
+        model_dict['y_min'] = 0.
```
Comment (tcuongd) [inline python/prophet/serialize.py:162]: I might be missing something, but is there a reason we need to move this logic up here? I don't think we call / want to call any of the methods setup methods (setup_dataframe, initilize_scales) during deserialization.

Could we instead:

* add a method that handles backwards compatibility for SIMPLE_ATTRIBUTES
* put your logic in lines 160-162 in that method?

There's another PR, https://github.com/facebook/prophet/pull/2477/files, that would benefit from this too.

**Review by tcuongd — COMMENTED** [review]
Thanks for the thoughtful and impactful contribution, as always! Really great tests too.

I just have a comment about `serialize.py`, let me know if I'm missing something.

Re: other types of scaling, I don't think we should worry about that for now. Seems like there's a lot of moving parts re: fit, predict, and serialize; if we want to allow more custom models in general it'll be a bigger effort.

File: python/prophet/serialize.py (hunk 1 above)
Comment (yoziru) [reply python/prophet/serialize.py]: Yep, you're right, we don't call it so this moving around was not necessary. 

This is a lingering change, remembered I had an iteration trying to infer the scale from `df.history` if it wasn't set (using `minmax` as default). 

Added separate method for backwards compat 👍

**Review by tcuongd — APPROVED** [review]
Thanks so much again, @yoziru !

**Pull request merged**